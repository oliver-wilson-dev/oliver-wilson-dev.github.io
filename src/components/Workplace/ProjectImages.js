import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const ProjectImages = ({ projectImages }) => (
  <div className={styles.projectImages}>
    {projectImages.map(({ linkToProject, image: { alt, src } }) => (
      <div key={linkToProject} className={styles.projectImage}>
        <a href={linkToProject}>
          <img alt={alt} src={src} />
        </a>
      </div>
    ))}
  </div>
);

ProjectImages.propTypes = {
  projectImages: PropTypes.arrayOf(
    PropTypes.shape({
      linkToProject: PropTypes.string,
      image: PropTypes.shape({
        alt: PropTypes.string,
        src: PropTypes.string
      })
    })
  )
};

export default ProjectImages;
