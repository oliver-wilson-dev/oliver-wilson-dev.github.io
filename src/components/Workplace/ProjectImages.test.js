import React from 'react';
import { shallow } from 'enzyme';
import ProjectImages from './ProjectImages';

const projectImage = {
  linkToProject: 'test-link-to-project',
  image: {
    alt: 'test-alt',
    src: 'test-src'
  }
};

const defaultProps = {
  projectImages: [projectImage]
};

const render = ({ overrideProps = {} } = { overrideProps: {} }) => shallow(
  <ProjectImages
    {...defaultProps}
    {...overrideProps}
  />
);

describe('ProjectImages', () => {
  describe('when there is one project image', () => {
    it('should render correctly', () => {
      expect(render()).toMatchSnapshot();
    });
  });

  describe('when there are multiple project images', () => {
    it('should render correctly', () => {
      expect(render({
        overrideProps: {
          projectImages: [projectImage, projectImage]
        }
      })).toMatchSnapshot();
    });
  });
});
