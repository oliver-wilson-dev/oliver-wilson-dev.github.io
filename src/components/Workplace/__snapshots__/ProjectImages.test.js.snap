// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectImages when there are multiple project images should render correctly 1`] = `
<div
  className="projectImages"
>
  <div
    className="projectImage"
    key="test-link-to-project"
  >
    <a
      href="test-link-to-project"
    >
      <img
        alt="test-alt"
        src="test-src"
      />
    </a>
  </div>
  <div
    className="projectImage"
    key="test-link-to-project"
  >
    <a
      href="test-link-to-project"
    >
      <img
        alt="test-alt"
        src="test-src"
      />
    </a>
  </div>
</div>
`;

exports[`ProjectImages when there is one project image should render correctly 1`] = `
<div
  className="projectImages"
>
  <div
    className="projectImage"
    key="test-link-to-project"
  >
    <a
      href="test-link-to-project"
    >
      <img
        alt="test-alt"
        src="test-src"
      />
    </a>
  </div>
</div>
`;
